### PR TITLE
WIP: Add i18n

### PR DESCRIPTION
**Pull request by astagi**
This PR adds i18n support. It's a work in progress, cause it needs some things to be completed:

- [x] Some hard coded labels to translate
- [ ] Define a list of languages the app should support
- [x] Understand how to get the correct `view` and `edit` urls based on language. The current interpolation under `off.js` doesn't work for some languages (e.g. [https://world-en.openfoodfacts.org/cgi/product.pl?type=edit&code=26041254](https://world-en.openfoodfacts.org/cgi/product.pl?type=edit&code=26041254) gives an error)

### CODE REVIEW

File: src/views/SettingsView.vue [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
-    const settings = localSettings.fetch();
-    return {
-      lang: settings.lang || "en"
-    };
+    return { langs: Object.keys(messages) }
   },
   methods: {
     saveLang: function() {
-      localSettings.update("lang", this.lang);
+      localSettings.update("lang", this.$i18n.locale);
```
Comment (raphael0202) [inline src/views/SettingsView.vue:32]: The language is updated as soon as we switch it, better to wait for a click on the save button

File: src/i18n/messages.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import it from './it'
+import en from './en'
+
+
+export default {
+  en: en,
```
Comment (raphael0202) [inline src/i18n/messages.js]: export default {
  en,
  it,
} works well too

File: src/i18n/en.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+export default  {
+  label: {
```
Comment (raphael0202) [inline src/i18n/en.js]: Missing pages: /logos, /logos/search, /insights

File: src/components/AnnotationCounter.vue [hunk 4]
```diff
@@ -1,16 +1,16 @@
 <template>
   <div class="main-column">
-    <h3>Annotations</h3>
+    <h3>{{$t('label.annotations')}}</h3>
     <p>
-      Remaining:
+      {{$t('label.remaining_annotations')}}:
```
Comment (raphael0202) [inline src/components/AnnotationCounter.vue]: it's better to add the ':' to the string to translate, as there may be a space needed after the word ("Remaining:" in english but "Restantes :" in French

**Review by raphael0202 — CHANGES_REQUESTED** [review]
Thanks for the PR 👍, I've added a few comments

**Comment by raphael0202** [comment]
Sorry I didn't see this was a WIP

File: src/components/AnnotationCounter.vue (hunk 4 above)
Comment (astagi) [reply src/components/AnnotationCounter.vue]: Thanks! didn't know that :)

**Comment by astagi** [comment]
No problem @raphael0202 I updated the PR with new translations and the requested changes

**Comment by astagi** [comment]
@raphael0202 I marked "Understand how to get the correct view and edit urls based on language." as done, looks like @teolemon is working to fix the problem in #57

**Comment by teolemon** [comment]
Erm I'm very unfamilar with Vue.JS. Feel free to fix the issue in this pr
image

**Comment by teolemon** [comment]
But the PR already LGTM to me

**Comment by astagi** [comment]
@teolemon that's good ✌🏻 Also if 'en' is the only case where `view` and `edit` links fail, you can write that part of code with a single line expression

```js
`https://world${lang === 'en' ? '' : '-' + lang}.openfoodfacts.org/product/${barcode}`
```

**Comment by raphael0202** [comment]
Sorry for the delay. A few translations are missing in en.json for the logo search page, but I will add them manually. Merging, thank you!

**Review by raphael0202 — APPROVED** [review]

**Comment by astagi** [comment]
Thanks @raphael0202 👍  sorry for missing translations, I didn't see them 👀  I'll check the missing item "Define a list of languages the app should support", supported languages can be added manually later

**Pull request merged**